enhancements/implicit-catalog-versioning.md

```diff
@@ -0,0 +1,190 @@
+│   └── etcdrestore.crd.yaml
+└── metadata
+    └── annotations.yaml
+```
+
+In this format, it becomes significantly less clear how to define your update graph. If you are only ever publishing a specific version of your operator, how do you know how it fits into the graph? What has even been published previously? Where do you even get the metadata needed to define your update graph? The author of this operator for this trivial example would have to either piece together their previous releases based on a hopefully accurate source control history, or they would need to use some external tooling (which does not currently exist today) to inspect the already created Index image to see what has been released and define their graph that way.
+
+The primary motivation for this enhancement is to simplify all of that complexity and guesswork by redefining the main driver for the update graph to just use semantic versioning. If the versioning is implicit (i.e. 1.3.1 is an update from 1.3.0) all the operator author needs to be aware of is "is my new version the latest thing?" -- they do not need to inspect the manifests of existing versions to construct the graph.
+
+In the future, this proposal also helps to solve the ordering problem when building and modifying index images. Today when adding to Index images via the `opm index` command, you need to specify your images in the order that they replace previous images because of the possibility that the explicit data will make an invalid intermediate graph. With the use of semantic version, it becomes trivial to insert elements into the middle of the graph. It also allows a future in which specific versions can be deleted from graphs without a need to edit the yaml.
+
+### Goals
+
+* Define Catalog update graphs for individual operators based implicitly on semantic version rather than asking users to explicitly mention what version the previous version replaces
+* Update OLM to handle the new format of catalogs
+* Handle conversion of existing operator subscriptions to use the new format
+
+### Non-Goals
+
+* Modify any of the features of OLM or the Operator Registry to use new information from the semantic version. This proposal simply suggests using the semantic version to define the graph
+* Resolve the migration of existing subscriptions to use the new Catalog Index images
+* Make any explicit changes to existing OLM API versions (CSV will remain `v1alpha1`)
+
+## Proposal
+
+Basic approach:
+
+- Make the `spec.version` field required going forward when building registry images with `opm registry` commands
+- Modify the operator registry to build dependency graph based simply on `spec.version` field rather than using a combination of channel heads and `spec.replaces` for each graph
+- Modify OLM to use the `spec.version` field everywhere it attempts to parse the graph for update information
+- Update skip feature to use version strings
+
+### Deprecate the `replaces` field:
```

Comment:
As per previous comment, I don't think we want to deprecate the replaces field; we should instead not require it for opm build commands. Removing the deprecation requirement simplifies the implementation and testing requirements for this enhancement (restricts it to one change in catalog-operator + opm load changes vs redoing and testing the upgrade flows in olm-operator)

I'm not saying we should never deprecate it, I'm just saying that for the goals outlined here of building an update graph based on SemVer and simplifying things for operator authors, it's not needed.

Reply:
Sorry, this header should just also be more clear. The intention of this enhancement is to just make it a requirement with `opm` and this doesn't explicitly deprecate `replaces` in the CRD definition (or anywhere in OLM itself).